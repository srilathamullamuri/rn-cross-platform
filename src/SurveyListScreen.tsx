/***********************************************************************EMCL2022
 *
 * SurveyListScreen.tsx
 *
 * NOTICE:  All information contained herein is, and remains
 * the property of EMC Software and its suppliers,
 * if any.  The intellectual and technical concepts contained
 * herein are proprietary to EMC Software
 * and its suppliers and may be covered by U.S. and Foreign Patents,
 * patents in process, and are protected by trade secret or copyright law.
 * Dissemination of this information or reproduction of this material
 * is strictly forbidden unless prior written permission is obtained
 * from EMC Software.
 *
 * @flow
 *
 * EMC Software Confidential, All Rights Reserved
 * Copyright (c) 2022, EMC Software
 *
 ***************************************************************************/

import * as React from 'react';
import {View, Text} from 'react-native';
import ReactTooltip from 'react-tooltip';
// import {ScaledSheet as StyleSheet} from 'react-native-size-matters';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {Table, TableWrapper, Row, Cell} from './components/Table';
import {ModalFile} from './components/ModalFile';
import {styles} from './styles';

// import styled from 'styled-components';
// import {useTable} from 'react-table';

export interface Props {
  list: any;
  showSurveyDetails: (item: any, surveyAnswerItem?: any) => void;
  showSurveyAnswerHistory: (item: any) => void;
  showPdf: (id: number) => void;
  reviseSurvey: (id: number) => void;
  modalFileUrl: string | undefined;
  // startNewResponse: (item: any) => void;
}

// eslint-disable-next-line react/prop-types
export const SurveyListScreen = (props: Props) => {
  const {modalFileUrl} = props;
  const headers = ['Assessment', 'Version', 'Actions'];
  const answerHeaders = ['Assessment Date', 'Status', 'Edit / View'];

  console.log('srilatha list', props?.list);

  return (
    <View style={localStyles.container}>
      {modalFileUrl && (
        <ModalFile
          url={modalFileUrl}
          onXClick={() => false}
          styles={{
            modStyle: styles.docCustomModelStyles,
            jpgStyle: styles.imagePDF,
          }}
          viewFileType="pdf"
        />
      )}
      <Table
        borderStyle={{borderWidth: 2, borderColor: '#c8e1ff'}}
        style={{width: '100%'}}>
        <Row
          data={headers}
          style={localStyles.head}
          textStyle={localStyles.text}
        />
        {props?.list?.map((item, i) => {
          return (
            <View style={{borderColor: '#c8e1ff', borderWidth: 1}} key={i}>
              <TableWrapper key={i} style={localStyles.row}>
                <Cell data={item.name} />
                <Cell data={item.version} />
                <Cell>
                  <View
                    style={{
                      flex: 3,
                      flexDirection: 'row',
                      padding: 10,
                      justifyContent: 'center',
                    }}>
                    {item.answerSetId && item.final && (
                      <div data-tip={'View'}>
                        <ReactTooltip />
                        <FontAwesomeIcon
                          icon="eye"
                          color={colors.ochBlue}
                          onClick={() =>
                            props.showSurveyDetails(item, {
                              id: item.answerSetId,
                              bFinal: item.final,
                            })
                          }
                          style={localStyles.marginSides}
                        />
                      </div>
                    )}
                    {item.answerSetId && !item.final && (
                      <div data-tip={'Edit'}>
                        <ReactTooltip />
                        <FontAwesomeIcon
                          icon="edit"
                          color={colors.ochBlue}
                          onClick={() =>
                            props.showSurveyDetails(item, {
                              id: item.answerSetId,
                              bFinal: item.final,
                            })
                          }
                          style={localStyles.marginSides}
                        />
                      </div>
                    )}
                    <div data-tip={'History'}>
                      <ReactTooltip />
                      <FontAwesomeIcon
                        icon="history"
                        color={colors.ochBlue}
                        onClick={() => props.showSurveyAnswerHistory(item)}
                        style={localStyles.marginSides}
                      />
                    </div>
                    {item.answerSetId && item.final && (
                      <div data-tip={'Revise'}>
                        <ReactTooltip />
                        <FontAwesomeIcon
                          icon="tools"
                          color={colors.ochBlue}
                          style={localStyles.marginSides}
                          onClick={() => props.reviseSurvey(item.answerSetId)}
                        />
                      </div>
                    )}
                    {item.answerSetId && (
                      <div data-tip={'Report'}>
                        <ReactTooltip />
                        <FontAwesomeIcon
                          icon="file-pdf"
                          color={colors.ochBlue}
                          onClick={() => props.showPdf(item.answerSetId)}
                          style={localStyles.marginSides}
                        />
                      </div>
                    )}
                    {((item.answerSetId && item.final) ||
                      item.answerSetId === undefined) && (
                      <div data-tip={'New'}>
                        <ReactTooltip />
                        {((item.answerSetId && item.final) ||
                          item.answerSetId === undefined) && (
                          <FontAwesomeIcon
                            icon="plus-circle"
                            color={colors.ochBlue}
                            onClick={() => props.showSurveyDetails(item)}
                            style={localStyles.marginSides}
                          />
                        )}
                      </div>
                    )}
                  </View>
                </Cell>
              </TableWrapper>
              {item.surveyAnswerSet?.length > 0 && (
                <View style={{padding: '1%'}}>
                  <Table
                    borderStyle={{borderWidth: 2, borderColor: '#c8e1ff'}}
                    style={{width: '100%'}}>
                    <Row
                      data={answerHeaders}
                      style={localStyles.head}
                      textStyle={localStyles.text}
                    />
                    {item.surveyAnswerSet
                      ?.sort((a, b) => {
                        const aa: any = new Date(a.revision);
                        const bb: any = new Date(b.revision);
                        return aa - bb;
                      })
                      .reverse()
                      .map((surveyAnswerItem, i_) => {
                        const theDate = new Date(surveyAnswerItem.revision);
                        return (
                          <TableWrapper key={i_} style={localStyles.row}>
                            <Cell data={theDate.toLocaleString('en-US')} />
                            <Cell data={surveyAnswerItem.status} />
                            <Cell>
                              <View
                                style={{
                                  flex: 2,
                                  flexDirection: 'row',
                                  padding: 10,
                                  justifyContent: 'center',
                                }}>
                                {!surveyAnswerItem.bFinal && (
                                  <div data-tip={'Edit'}>
                                    <ReactTooltip />
                                    <FontAwesomeIcon
                                      icon="edit"
                                      color={colors.ochBlue}
                                      onClick={() =>
                                        props.showSurveyDetails(
                                          item,
                                          surveyAnswerItem,
                                        )
                                      }
                                      style={localStyles.marginSides}
                                    />
                                  </div>
                                )}  
                                {surveyAnswerItem.bFinal && (
                                  <div data-tip={'View'}>
                                    <ReactTooltip />
                                    <FontAwesomeIcon
                                      icon="eye"
                                      color={colors.ochBlue}
                                      onClick={() =>
                                        props.showSurveyDetails(
                                          item,
                                          surveyAnswerItem,
                                        )
                                      }
                                      style={localStyles.marginSides}
                                    />
                                  </div>
                                )}
                                {surveyAnswerItem.bFinal && (
                                  <div data-tip={'Revise'}>
                                    <ReactTooltip />
                                    <FontAwesomeIcon
                                      icon="tools"
                                      color={colors.ochBlue}
                                      style={localStyles.marginSides}
                                      onClick={() => props.reviseSurvey(surveyAnswerItem.id)}
                                    />
                                  </div>
                                )}
                                <div data-tip={'Report'}>
                                  <ReactTooltip />
                                  <FontAwesomeIcon
                                    icon="file-pdf"
                                    color={colors.ochBlue}
                                    onClick={() => props.showPdf(surveyAnswerItem.id)}
                                    style={localStyles.marginSides}
                                  />
                                </div>
                              </View>
                            </Cell>
                          </TableWrapper>
                        );
                      })}
                  </Table>
                </View>
              )}
              {item.surveyAnswerSet?.length <= 0 && (
                <View>
                  <Text style={{alignSelf: 'center'}}>No Document History</Text>
                </View>
              )}
            </View>
          );
        })}
      </Table>
    </View>
  );
};

const colors = {
  red: '#FC4349',
  green: '#8CC63F',
  //   orange: '#ffa500',

  white: '#ffffff',
  lighter: '#F1F5FC',
  light: '#dee4e7',
  dark: '#777777',
  darker: '#213249',
  black: '#000000',

  background: '#ffffff',
  border: '#ced4da',
  placeHolderText: '#555555',
  error: '#FC4349',
  alert: '#FC4349',
  valid: '#009900',
  textBlack: '#000000',

  base: '#445878',
  primary: '#799AE0',
  pink: '#FF87C3',
  grey: '#9dafc8',
  darkGrey: '#202020',
  darkBlue: '#1B4581',
  cash: '#FF87C3',
  // nzSecurities: '#799AE0',
  property: '#92CDCF',
  // nzEquities: '#8CC63F',
  // ausEquities: '#DECD58',
  // globalEquities: '#FFAA4A',
  // altStrategies: '#FC4349',
  // globalDebtSecurities: '#213249',
  // from toolbox
  turquoise: '#1abc9c',
  // emerald: '#2ecc71',
  peterRiver: '#3498db',
  // amethyst: '#9b59b6',
  // wetAsphalt: '#34495e',
  greenSea: '#16a085',
  nephritis: '#27ae60',
  belizeHole: '#2980b9',
  wisteria: '#8e44ad',
  // midnightBlue: '#2c3e50',
  sunFlower: '#f1c40f',
  // carrot: '#e67e22',
  // alizarin: '#e74c3c',
  // clouds: '#ecf0f1',
  // concrete: '#95a5a6',
  orange: '#f39c12',
  pumpkin: '#d35400',
  // pomegranate: '#c0392b',
  silver: '#bdc3c7',
  asbestos: '#7f8c8d',

  ochActivityIndicator: '#f09b88',

  ochBlue: '#237aff', // #4073bf
  ochBG: '#579AFF',
  //Triadic Colors of #3578e5
  ochBlueTri2: '#e53578',
  ochBlueTri3: '#78e535',
  // Analogous Colors of #3578e5
  ochBlueA1: '#4a35e5',
  ochBlueA2: '#35d0e5',
  // Monochromatic Colors of #3578e5',
  ochBlueM3: '#1754b6',
  ochBlueM2: '#1a5ecd',
  ochBlueM1: '#1e69e2',
  // ochBlueM4: '#3578e5',
  ochBlueP1: '#4c87e8',
  ochBlueP2: '#6296eb',
  ochBlueP3: '#79a5ee',
  // Complementary Color
  ochBlueComp: '#e5a235',
  ochRedBG: '#ec71a0',
  buttonText: '#000000',
  buttonBorder: '#c5c5cf',
};

const localStyles = {
  container: {
    flex: 1,
    padding: 10,
    width: '80vw',
  },
  tableColumnClockInOutTimes: {
    alignItems: 'center',
    flex: 3,
    justifyContent: 'center',
    borderWidth: 1,
  },
  tableRow: {
    flex: 5,
    borderWidth: 1,
    borderStyle: 'solid',
    borderColor: 'black',
  },
  tableContainer: {
    borderRadius: 5,
    flex: 1,
    marginTop: 0,
    padding: 10,
  },
  textHeader: {
    fontWeight: 'bold',
  },
  textLineItem: {
    color: 'black',
  },
  marginSides: {
    marginLeft: 10,
    marginRight: 10,
  },
  head: {height: 40, backgroundColor: '#f1f8ff'},
  text: {margin: 6},
  row: {flexDirection: 'row', backgroundColor: '#FFF1C1'},
};

export default SurveyListScreen;
