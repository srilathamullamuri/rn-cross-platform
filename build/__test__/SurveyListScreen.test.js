import React from 'react';
import renderer from 'react-test-renderer';
import SurveyListScreen from '../src/SurveyListScreen';
test('renders correctly Survey List Screen', function () {
    var data = [
        { id: 1, name: 'Life Plan', version: '0.9', answerSetId: 76, final: false },
    ];
    var tree = renderer.create(React.createElement(SurveyListScreen, { list: data })).toJSON();
    expect(tree).toMatchSnapshot();
});
//# sourceMappingURL=SurveyListScreen.test.js.map