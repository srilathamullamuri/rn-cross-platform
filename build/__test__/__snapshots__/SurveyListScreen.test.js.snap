// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly Survey List Screen 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitFlex": 1,
      "flex": 1,
      "msFlex": "1 1 0%",
      "paddingBottom": "10px",
      "paddingLeft": "10px",
      "paddingRight": "10px",
      "paddingTop": "10px",
      "width": "80vw",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomColor": "rgba(200,225,255,1.00)",
        "borderBottomWidth": "2px",
        "borderLeftColor": "rgba(200,225,255,1.00)",
        "borderLeftWidth": "2px",
        "borderRightColor": "rgba(200,225,255,1.00)",
        "borderTopColor": "rgba(200,225,255,1.00)",
        "width": "100%",
      }
    }
  >
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "WebkitBoxDirection": "normal",
          "WebkitBoxOrient": "horizontal",
          "WebkitFlexDirection": "row",
          "backgroundColor": "rgba(241,248,255,1.00)",
          "flexDirection": "row",
          "height": "40px",
          "msFlexDirection": "row",
          "overflowX": "hidden",
          "overflowY": "hidden",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxPack": "center",
            "WebkitFlex": 1,
            "WebkitJustifyContent": "center",
            "borderBottomColor": "rgba(0,0,0,1.00)",
            "borderLeftColor": "rgba(0,0,0,1.00)",
            "borderRightColor": "rgba(0,0,0,1.00)",
            "borderRightWidth": "0px",
            "borderTopColor": "rgba(0,0,0,1.00)",
            "borderTopWidth": "0px",
            "flex": 1,
            "justifyContent": "center",
            "msFlex": "1 1 0%",
            "msFlexPack": "center",
            "paddingBottom": "5px",
            "paddingLeft": "5px",
            "paddingRight": "5px",
            "paddingTop": "5px",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.00)",
              "marginBottom": "6px",
              "marginLeft": "6px",
              "marginRight": "6px",
              "marginTop": "6px",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
              "textAlign": "center",
            }
          }
        >
          Assessment
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxPack": "center",
            "WebkitFlex": 1,
            "WebkitJustifyContent": "center",
            "borderBottomColor": "rgba(0,0,0,1.00)",
            "borderLeftColor": "rgba(0,0,0,1.00)",
            "borderRightColor": "rgba(0,0,0,1.00)",
            "borderRightWidth": "0px",
            "borderTopColor": "rgba(0,0,0,1.00)",
            "borderTopWidth": "0px",
            "flex": 1,
            "justifyContent": "center",
            "msFlex": "1 1 0%",
            "msFlexPack": "center",
            "paddingBottom": "5px",
            "paddingLeft": "5px",
            "paddingRight": "5px",
            "paddingTop": "5px",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.00)",
              "marginBottom": "6px",
              "marginLeft": "6px",
              "marginRight": "6px",
              "marginTop": "6px",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
              "textAlign": "center",
            }
          }
        >
          Version
        </div>
      </div>
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxPack": "center",
            "WebkitFlex": 1,
            "WebkitJustifyContent": "center",
            "borderBottomColor": "rgba(0,0,0,1.00)",
            "borderLeftColor": "rgba(0,0,0,1.00)",
            "borderRightColor": "rgba(0,0,0,1.00)",
            "borderRightWidth": "0px",
            "borderTopColor": "rgba(0,0,0,1.00)",
            "borderTopWidth": "0px",
            "flex": 1,
            "justifyContent": "center",
            "msFlex": "1 1 0%",
            "msFlexPack": "center",
            "paddingBottom": "5px",
            "paddingLeft": "5px",
            "paddingRight": "5px",
            "paddingTop": "5px",
          }
        }
      >
        <div
          className="css-text-901oao"
          dir="auto"
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,0.00)",
              "marginBottom": "6px",
              "marginLeft": "6px",
              "marginRight": "6px",
              "marginTop": "6px",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
              "textAlign": "center",
            }
          }
        >
          Actions
        </div>
      </div>
    </div>
    <div
      className="css-view-1dbjc4n"
      style={
        Object {
          "borderBottomColor": "rgba(200,225,255,1.00)",
          "borderBottomWidth": "1px",
          "borderLeftColor": "rgba(200,225,255,1.00)",
          "borderLeftWidth": "1px",
          "borderRightColor": "rgba(200,225,255,1.00)",
          "borderRightWidth": "1px",
          "borderTopColor": "rgba(200,225,255,1.00)",
          "borderTopWidth": "1px",
        }
      }
    >
      <div
        className="css-view-1dbjc4n"
        style={
          Object {
            "WebkitBoxDirection": "normal",
            "WebkitBoxOrient": "horizontal",
            "WebkitFlexDirection": "row",
            "backgroundColor": "rgba(255,241,193,1.00)",
            "flexDirection": "row",
            "msFlexDirection": "row",
          }
        }
      >
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxPack": "center",
              "WebkitFlex": 1,
              "WebkitJustifyContent": "center",
              "borderBottomColor": "rgba(0,0,0,1.00)",
              "borderLeftColor": "rgba(0,0,0,1.00)",
              "borderRightColor": "rgba(0,0,0,1.00)",
              "borderRightWidth": "0px",
              "borderTopColor": "rgba(0,0,0,1.00)",
              "borderTopWidth": "0px",
              "flex": 1,
              "justifyContent": "center",
              "msFlex": "1 1 0%",
              "msFlexPack": "center",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "paddingBottom": "5px",
                "paddingLeft": "5px",
                "paddingRight": "5px",
                "paddingTop": "5px",
                "textAlign": "center",
              }
            }
          >
            Life Plan
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxPack": "center",
              "WebkitFlex": 1,
              "WebkitJustifyContent": "center",
              "borderBottomColor": "rgba(0,0,0,1.00)",
              "borderLeftColor": "rgba(0,0,0,1.00)",
              "borderRightColor": "rgba(0,0,0,1.00)",
              "borderRightWidth": "0px",
              "borderTopColor": "rgba(0,0,0,1.00)",
              "borderTopWidth": "0px",
              "flex": 1,
              "justifyContent": "center",
              "msFlex": "1 1 0%",
              "msFlexPack": "center",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
            }
          }
        >
          <div
            className="css-text-901oao"
            dir="auto"
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.00)",
                "paddingBottom": "5px",
                "paddingLeft": "5px",
                "paddingRight": "5px",
                "paddingTop": "5px",
                "textAlign": "center",
              }
            }
          >
            0.9
          </div>
        </div>
        <div
          className="css-view-1dbjc4n"
          style={
            Object {
              "WebkitBoxPack": "center",
              "WebkitFlex": 1,
              "WebkitJustifyContent": "center",
              "borderBottomColor": "rgba(0,0,0,1.00)",
              "borderLeftColor": "rgba(0,0,0,1.00)",
              "borderRightColor": "rgba(0,0,0,1.00)",
              "borderRightWidth": "0px",
              "borderTopColor": "rgba(0,0,0,1.00)",
              "borderTopWidth": "0px",
              "flex": 1,
              "justifyContent": "center",
              "msFlex": "1 1 0%",
              "msFlexPack": "center",
              "paddingBottom": "5px",
              "paddingLeft": "5px",
              "paddingRight": "5px",
              "paddingTop": "5px",
            }
          }
        >
          <div
            className="css-view-1dbjc4n"
            style={
              Object {
                "WebkitBoxDirection": "normal",
                "WebkitBoxOrient": "horizontal",
                "WebkitBoxPack": "center",
                "WebkitFlex": 3,
                "WebkitFlexDirection": "row",
                "WebkitJustifyContent": "center",
                "flex": 3,
                "flexDirection": "row",
                "justifyContent": "center",
                "msFlex": "3 1 0%",
                "msFlexDirection": "row",
                "msFlexPack": "center",
                "paddingBottom": "10px",
                "paddingLeft": "10px",
                "paddingRight": "10px",
                "paddingTop": "10px",
              }
            }
          >
            <div
              data-tip="Edit"
            >
              <div
                className="__react_component_tooltip t226ab1e8-a20f-44f0-8068-0c680ed6f33b place-top type-dark"
                data-id="tooltip"
                id="t226ab1e8-a20f-44f0-8068-0c680ed6f33b"
              >
                <style
                  aria-hidden="true"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "
  	.t226ab1e8-a20f-44f0-8068-0c680ed6f33b {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-top {
        margin-top: -10px;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-top::before {
        border-top: 8px solid transparent;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-bottom {
        margin-top: 10px;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-left {
        margin-left: -10px;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-left::before {
        border-left: 8px solid transparent;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-right {
        margin-left: 10px;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-right::before {
        border-right: 8px solid transparent;
    }
    .t226ab1e8-a20f-44f0-8068-0c680ed6f33b.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
                    }
                  }
                />
              </div>
            </div>
            <div
              data-tip="History"
            >
              <div
                className="__react_component_tooltip td1c195ac-2217-487a-b13f-e9c0a55e010f place-top type-dark"
                data-id="tooltip"
                id="td1c195ac-2217-487a-b13f-e9c0a55e010f"
              >
                <style
                  aria-hidden="true"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "
  	.td1c195ac-2217-487a-b13f-e9c0a55e010f {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.td1c195ac-2217-487a-b13f-e9c0a55e010f.place-top {
        margin-top: -10px;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-top::before {
        border-top: 8px solid transparent;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-bottom {
        margin-top: 10px;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-left {
        margin-left: -10px;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-left::before {
        border-left: 8px solid transparent;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-right {
        margin-left: 10px;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-right::before {
        border-right: 8px solid transparent;
    }
    .td1c195ac-2217-487a-b13f-e9c0a55e010f.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
                    }
                  }
                />
              </div>
            </div>
            <div
              data-tip="Report"
            >
              <div
                className="__react_component_tooltip t82dea625-6a71-424a-9076-5718385566f5 place-top type-dark"
                data-id="tooltip"
                id="t82dea625-6a71-424a-9076-5718385566f5"
              >
                <style
                  aria-hidden="true"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "
  	.t82dea625-6a71-424a-9076-5718385566f5 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t82dea625-6a71-424a-9076-5718385566f5.place-top {
        margin-top: -10px;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-top::before {
        border-top: 8px solid transparent;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t82dea625-6a71-424a-9076-5718385566f5.place-bottom {
        margin-top: 10px;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t82dea625-6a71-424a-9076-5718385566f5.place-left {
        margin-left: -10px;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-left::before {
        border-left: 8px solid transparent;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t82dea625-6a71-424a-9076-5718385566f5.place-right {
        margin-left: 10px;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-right::before {
        border-right: 8px solid transparent;
    }
    .t82dea625-6a71-424a-9076-5718385566f5.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
